### PR TITLE
Arm support 2625

### LINKED ISSUE #2625: Support Apple M1 devices
As a user of desktop mac application i want to have support for arm architecture so i don't have to enable legacy support for my applie silicon mac.

## AC
* User can use the desktop application without rosetta
* Intel macs are still supported
* There is one binary which supports both architectures.

## Tasks
- [x] Test whether running `node desktop -a universal` builds a universal binary on Mac
- [x] Adjust Jenkins job to build a universal binary on M1 mac
- [x] Try merge option
- [x] Test whether the desktop client runs on older versions of macOS (BigSur & Monterey)
 
## Notes

* We might need to setup the build environment on the new mac if we cannot build apple silicon binaries on the old one.

* We need to include arm precompiled binaries for openssl/sql cipher.

## Notes (issue reported by user)

**Describe the bug**
There is no arm64 binary available for Apple M1 devices. Currently the app needs to be run through Rosetta translation. This is important as all Apple desktop/laptop devices will be using this chip going forward and they *claim* it's as easy as adding an additional target in the build process. 

**To Reproduce**
NA

**Expected behavior**
Native binary available for Apple Silicon (currently only M1). 

**Screenshots**
2021-01-16 at 10 20

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version: 3.80.4

**Additional context**
http://doesitarm.com shows a bunch of apps which have already implemented native support.

## Test Notes
 * [ ] the dmg does work on MacOS 
   * on the M1 chip
   * also for intel
* [ ] linux desktop client starts
* [ ] on all desktop platform, it's possible to use previously stored credentials to log in

### LINKED ISSUE #6050: Segfault on old macOS versions
**Workaround**: Please try downgrading to the version mentioned in this comment:

https://github.com/tutao/tutanota/issues/6050#issuecomment-1816180959

v3.118.22

```
Process:               Tutanota Desktop [668]
Path:                  /Applications/Tutanota [Desktop.app/Contents/MacOS/Tutanota](http://desktop.app/Contents/MacOS/Tutanota) Desktop
Identifier:            de.tutao.tutanota
Version:               3.118.22 (3.118.22.611)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Tutanota Desktop [668]
User ID:               501

Date/Time:             2023-10-27 07:16:10.048 +0200
OS Version:            Mac OS X 10.13.6 (17G14042)
Report Version:        12
Anonymous UUID:        B808121E-5C3B-28BF-B9A8-8A318182F9D5

Sleep/Wake UUID:       123C3935-CB99-4C06-899D-BA79A64C85C2

Time Awake Since Boot: 440 seconds
Time Since Wake:       13 seconds

System Integrity Protection: enabled

Crashed Thread:        0  CrBrowserMain  Dispatch queue: [com.apple](http://com.apple/).main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

Thread 0 Crashed:: CrBrowserMain  Dispatch queue: [com.apple](http://com.apple/).main-thread
0   keytar.node                   0x0000000114bb7000 0x114bb7000 + 0
1   com.github.Electron.framework 0x000000010b5645c2 0x10b530000 + 214466
2   com.github.Electron.framework 0x000000010b562d67 0x10b530000 + 208231
3   com.github.Electron.framework 0x000000010b562d67 0x10b530000 + 208231
4   com.github.Electron.framework 0x000000010b666170 0x10b530000 + 1270128
5   com.github.Electron.framework 0x000000010b58dca7 0x10b530000 + 384167
6   com.github.Electron.framework 0x000000010b560b87 0x10b530000 + 199559
7   com.github.Electron.framework 0x000000010cd94d98 v8::internal::Execution::Call(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*) + 3960
8   com.github.Electron.framework 0x000000010cd95b8a v8::internal::Execution::TryCallScript(v8::internal::Isolate*, v8::internal::Handle<v8::internal::JSFunction>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::FixedArray>, v8::internal::Execution::MessageHandling, v8::internal::MaybeHandle<v8::internal::Object>*, bool) + 266
9   com.github.Electron.framework 0x000000010cd95cf9 v8::internal::Execution::TryCall(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*, v8::internal::Execution::MessageHandling, v8::internal::MaybeHandle<v8::internal::Object>*, bool) + 217
10  com.github.Electron.framework 0x000000010cdc95ef v8::internal::MicrotaskQueue::get(long) const + 479
11  com.github.Electron.framework 0x0000000112892020 node::CallbackScope::~CallbackScope() + 832
12  com.github.Electron.framework 0x0000000112891d3e node::CallbackScope::~CallbackScope() + 94
13  com.github.Electron.framework 0x000000011295aa66 node::OnFatalError(char const*, char const*) + 32870
14  com.github.Electron.framework 0x00000001129765b0 node::OnFatalError(char const*, char const*) + 146352
15  com.github.Electron.framework 0x000000011294dd02 node::Buffer::New(v8::Isolate*, char*, unsigned long) + 162770
16  com.github.Electron.framework 0x000000010b942b14 uv_random + 1428
17  com.github.Electron.framework 0x000000010b946be6 uv_async_send + 790
18  com.github.Electron.framework 0x000000010b958f29 uv_free_interface_addresses + 1817
19  com.github.Electron.framework 0x000000010b9470fa uv_run + 522
20  com.github.Electron.framework 0x000000010bae890a v8_inspector::V8DebuggerId::V8DebuggerId(std::__Cr::pair<long long, long long>) + 20698
21  com.github.Electron.framework 0x000000010ee65599 node::AsyncResource::get_async_id() const + 9454569
22  com.github.Electron.framework 0x000000010ee815fb node::AsyncResource::get_async_id() const + 9569355
23  com.github.Electron.framework 0x000000010ee81e95 node::AsyncResource::get_async_id() const + 9571557
24  com.github.Electron.framework 0x000000010bb7471e v8::internal::compiler::CodeAssemblerVariable::IsBound() const + 5166
25  com.github.Electron.framework 0x000000010b703c8a 0x10b530000 + 1916042
26  com.github.Electron.framework 0x000000010bb73c11 v8::internal::compiler::CodeAssemblerVariable::IsBound() const + 2337
27  [com.apple](http://com.apple/).CoreFoundation       0x00007fff3c350b01 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
28  [com.apple](http://com.apple/).CoreFoundation       0x00007fff3c408e5c __CFRunLoopDoSource0 + 108
29  [com.apple](http://com.apple/).CoreFoundation       0x00007fff3c333ab0 __CFRunLoopDoSources0 + 208
30  [com.apple](http://com.apple/).CoreFoundation       0x00007fff3c332f2d __CFRunLoopRun + 1293
31  [com.apple](http://com.apple/).CoreFoundation       0x00007fff3c332787 CFRunLoopRunSpecific + 487
32  [com.apple](http://com.apple/).HIToolbox           0x00007fff3b612d96 RunCurrentEventLoopInMode + 286
33  [com.apple](http://com.apple/).HIToolbox           0x00007fff3b612b06 ReceiveNextEventCommon + 613
34  [com.apple](http://com.apple/).HIToolbox           0x00007fff3b612884 _BlockUntilNextEventMatchingListInModeWithFilter + 64
35  [com.apple](http://com.apple/).AppKit               0x00007fff398bfa3b _DPSNextEvent + 2085
36  [com.apple](http://com.apple/).AppKit               0x00007fff3a055e34 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 3044
37  [com.apple](http://com.apple/).AppKit               0x00007fff398b484d -[NSApplication run] + 764
38  com.github.Electron.framework 0x000000010bb75466 v8::internal::compiler::CodeAssemblerVariable::IsBound() const + 8566
39  com.github.Electron.framework 0x000000010bb735f1 v8::internal::compiler::CodeAssemblerVariable::IsBound() const + 769
40  com.github.Electron.framework 0x000000010ee8215d node::AsyncResource::get_async_id() const + 9572269
41  com.github.Electron.framework 0x000000010ee47abf node::AsyncResource::get_async_id() const + 9333007
42  com.github.Electron.framework 0x000000010debb0c0 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 4122656
43  com.github.Electron.framework 0x000000010debca72 v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 4129234
44  com.github.Electron.framework 0x000000010deb8a1f v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 4112767
45  com.github.Electron.framework 0x000000010bc94c5d v8::internal::compiler::BasicBlock::set_loop_header(v8::internal::compiler::BasicBlock*) + 17053
46  com.github.Electron.framework 0x000000010bc95ca7 v8::internal::compiler::BasicBlock::set_loop_header(v8::internal::compiler::BasicBlock*) + 21223
47  com.github.Electron.framework 0x000000010bc95abf v8::internal::compiler::BasicBlock::set_loop_header(v8::internal::compiler::BasicBlock*) + 20735
48  com.github.Electron.framework 0x000000010bc94555 v8::internal::compiler::BasicBlock::set_loop_header(v8::internal::compiler::BasicBlock*) + 15253
49  com.github.Electron.framework 0x000000010bc94657 v8::internal::compiler::BasicBlock::set_loop_header(v8::internal::compiler::BasicBlock*) + 15511
50  com.github.Electron.framework 0x000000010b95982b ElectronMain + 155
51  libdyld.dylib                 0x00007fff64483015 start + 1
```

output is

```
[1032:1027/081743.101405:ERROR:mach_port_rendezvous.cc(347)] mach_msg: (ipc/send) invalid destination port (0x10000003)
zsh: segmentation fault  /Applications/Tutanota\ Desktop.app/Contents/MacOS/Tutanota\ Desktop
[1032:1027/081743.102247:ERROR:child_thread_impl.cc(230)] Mach rendezvous failed, terminating process (parent died?)
```

but it's probably because parent process has died because `mach_port_rendezvous` is part of chromium

## Test Notes
* [ ] the app does not crash on a Big Sur MacOs (does not need to show old credentials)
* [ ] also no crash on newer versions, as well as usable stored credentials

### PR DESCRIPTION
**Pull request by ganthern**
close #2625
close #6050
also related: https://github.com/tutao/tutanota/issues/1891

### CODE REVIEW

**Comment by rezbyte** [comment]
I've tested this on macOS Big Sur and it crashes on boot. From the console logs, I suspect another issue with entitlements.
`Sandbox: test tuta Mail D(699) deny(1) mach-lookup de.tutao.tutanota-test.MachPortRandezvousServer.1` seems to be the main error. The terminal gives `bootstrap_look_up de.tutao.tutanota-test.MachPortRandezvousServer.1: Permission denied (1100)`.

**Comment by rezbyte** [comment]
I looked up the terminal error, `1100` is the Mach error code for `BOOTSTRAP_NOT_PRIVILEGED`. Just tested the current release version on the older mac and it crashes too. Perhaps the crash is caused by the virtualization.

**Comment by rezbyte** [comment]
Tested it on a newer macOS, it worked fine. Seems there is still a problem with older versions.

File: src/desktop/sse/SecretStorage.ts [hunk 1]
```diff
@@ -33,8 +94,122 @@ export class KeytarSecretStorage implements SecretStorage {
...
+	/**
+	 * ensures that the safe_storage directory exists and that we can use the
+	 * safeStorage API
+	 * @private
+	 */
+	private async assertAvailable(): Promise<void> {
+		await this.electron.app.whenReady()
+		await this.fs.promises.mkdir(this.getSafeStoragePath(), { recursive: true })
+		if (this.electron.safeStorage.isEncryptionAvailable()) {
+			if (!this.initialized && process.platform === "linux") {
```
Comment (charlag) [inline src/desktop/sse/SecretStorage.ts:166]: you could use some kind of "once" thing like `lazyMemoized` to not print every time

Comment (ganthern) [reply src/desktop/sse/SecretStorage.ts:166]: isn't that functionally the same thing

Comment (charlag) [reply src/desktop/sse/SecretStorage.ts:166]: yup, just saying

**Comment by ganthern** [comment]
is this gtg or do we need the lazyMemoized

**Pull request merged**